pages/mesosphere/dcos/2.1/security/ent/tls-ssl/ar-custom/index.md

```diff
@@ -0,0 +1,243 @@
+
+## external_certificate_key_path
+Path (relative to the `$DCOS_INSTALL_DIR`) to a file containing the private key corresponding to the custom external certificate, encoded in the OpenSSL (PKCS#8) PEM format. For example: `genconf/external-certificate-key.key`.
+
+This is highly sensitive data. The configuration processor accesses this file only for configuration validation purposes, and does not copy the data. After successful configuration validation this file needs to be placed out-of-band into the file system of all DC/OS master nodes to the path `/var/lib/dcos/pki/tls/private/adminrouter-external.key` before most DC/OS `systemd` units start up. The file must be readable by the root user, and should have have 0600 permissions set.
+
+This path is required if `external_certificate_path` is specified.
+
+## external_certificate_servernames
+
+The list of server names that clients can use while accessing the cluster in
+order to make Admin Router present the custom external certificate to the
+client. The server names can include:
+  - a plain hostname, e.g. `host.example.com`.
+  - an IP address, e.g. `192.0.2.5`.
+  - a `*` wildcard, e.g. `*.example.com`, will match all the subdomains of
+    `example.com` (e.g. `foo.example.com`, `bar.example.com`), but not the domain
+    itself - `example.com` or deeper subdomains: `foo.bar.example.com`.
+  - a `.` wildcard, e.g. `.example.com`, will match the same as `*` wildcard plus the domain
+    itself (e.g. `foo.example.com`, `bar.example.com`, `example.com`).
+
+  The server names cannot include:
+  - `master.mesos`
+  - `leader.mesos`
+  - `registry.component.thisdcos.directory`
+
+Every server name provided in this option must be present in the custom external
+certificate's Canonical Name or Subject Alternative Names.
+
+This option is required if `external_certificate_path` is specified.
+
+## external_certificate_validation_disable
+The custom external certificate undergoes validation before being taken into
+consideration. Namely:
+    * the restrictions for server names mentioned for
+      `external_certificate_servernames` parameter
+    * certificate key must be of type RSA or ECC
+    * minimum size of an RSA key is 2048
+    * minimum size of the ECC key is 256
+    * the ECC's key curve must be one of SECP256R1 or SECP384R1,
+    * key usage extension must be defined
+    * the validity period of the certificate must not start in the future
+    * the validity period of the certificate must not end sooner than in 365
+      days
+    * the basic constraints are not required, but if they are defined, they
+      should not have the CA flag set
+    * key usage extension must have the digital signature and key encipherment
+      extensions set
+    * extended key usage extension must be present and tls server
+      authentication flag must be set
```

Comment:
The formatting for this list is messed up.   I think there are too many leading spaces before the asterisks, it does not show up as a list in the preview.

Reply:
Done